pyproject.toml

```diff
@@ -14,22 +14,25 @@ python = "^3.7"
 # Python lack of functionalities from future versions
 importlib-metadata = { version = "*", python = "<3.8" }
 
-neptune = ">=1.0.0"
 numpy = "<1.24.0"
 fvcore = "<0.1.5.post20221220"
 
 # dev
 pre-commit = { version = "*", optional = true }
 pytest = { version = ">=5.0", optional = true }
 pytest-cov = { version = "2.10.1", optional = true }
-torch = "^1.13.0"
-torchvision = "^0.14.0"
+torch = { version = "^1.13.0", optional = true }
+torchvision = { version = "^0.14.0", optional = true }
+neptune = { version = ">=1.0.0", optional = true }
```

Comment:
I think detectron2 should also be an optional dependency. Wdyt?

Reply:
It's not that easy, as it is only installed from GitHub repo, not PyPI and I've been having issued with adding that to poetry (that's why there is a seperate step just to install detectron2 in out CI)

Reply:
Makes sense.